pynamodb/tests/test_signals.py

```diff
@@ -0,0 +1,106 @@
+
+    pre_dynamo_send.connect(record_pre_dynamo_send)
+    post_dynamo_send.connect(record_post_dynamo_send)
+    try:
+        mock_uuid.uuid4.return_value = UUID
+        mock_req.return_value = {'TableDescription': {'TableName': 'table', 'TableStatus': 'Creating'}}
+        c = Connection()
+        c.dispatch('CreateTable', {'TableName': 'MyTable'})
+        assert ('CreateTable', 'MyTable', UUID) == pre_recorded[0]
+        assert ('CreateTable', 'MyTable', UUID) == post_recorded[0]
+    finally:
+        pre_dynamo_send.disconnect(record_pre_dynamo_send)
+        post_dynamo_send.disconnect(record_post_dynamo_send)
+
+
+@mock.patch(PATCH_METHOD)
+@mock.patch('pynamodb.connection.base.uuid')
+def test_signal_exception_pre_signal(mock_uuid, mock_req):
+    post_recorded = []
+    UUID = '123-abc'
+
+    def record_pre_dynamo_send(sender, operation_name, table_name, req_uuid):
+        raise ValueError()
+
+    def record_post_dynamo_send(sender, operation_name, table_name, req_uuid):
+        post_recorded.append((operation_name, table_name, req_uuid))
+
+    pre_dynamo_send.connect(record_pre_dynamo_send)
+    post_dynamo_send.connect(record_post_dynamo_send)
+    try:
+        mock_uuid.uuid4.return_value = UUID
+        mock_req.return_value = {'TableDescription': {'TableName': 'table', 'TableStatus': 'Creating'}}
+        c = Connection()
+        c.dispatch('CreateTable', {'TableName': 'MyTable'})
+        assert ('CreateTable', 'MyTable', UUID) == post_recorded[0]
+    finally:
+        pre_dynamo_send.disconnect(record_pre_dynamo_send)
+        post_dynamo_send.disconnect(record_post_dynamo_send)
+
+
+@mock.patch(PATCH_METHOD)
+@mock.patch('pynamodb.connection.base.uuid')
+def test_signal_exception_post_signal(mock_uuid, mock_req):
+    pre_recorded = []
+    UUID = '123-abc'
+
+    def record_pre_dynamo_send(sender, operation_name, table_name, req_uuid):
+        pre_recorded.append((operation_name, table_name, req_uuid))
+
+    def record_post_dynamo_send(sender, operation_name, table_name, req_uuid):
+        raise ValueError()
+
+    pre_dynamo_send.connect(record_pre_dynamo_send)
+    post_dynamo_send.connect(record_post_dynamo_send)
+    try:
+        mock_uuid.uuid4.return_value = UUID
+        mock_req.return_value = {'TableDescription': {'TableName': 'table', 'TableStatus': 'Creating'}}
+        c = Connection()
+        c.dispatch('CreateTable', {'TableName': 'MyTable'})
+        assert ('CreateTable', 'MyTable', UUID) == pre_recorded[0]
+    finally:
+        pre_dynamo_send.disconnect(record_pre_dynamo_send)
+        post_dynamo_send.disconnect(record_post_dynamo_send)
+
+
+def test_fake_signals():
+    _signals = _FakeNamespace()
+    pre_dynamo_send = _signals.signal('pre_dynamo_send')
+    with pytest.raises(RuntimeError):
+        pre_dynamo_send.connect(lambda x: x)
+    pre_dynamo_send.send(object, operation_name="UPDATE", table_name="TEST", req_uuid="something")
```

Comment:
@garrettheel i test the case where blinker is not installed this way. Where I just use the _FakeNamespace and make sure no exceptions happen when we call `send` but when you try to connect to it, it causes a runtime error.